### PR TITLE
Implement (optional) ZeroBuf support

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
retest this please

File: tests/connection/broker.cpp [hunk 1]
```diff
@@ -52,7 +56,11 @@ class Subscriber
...
         {
             if( subscriber.receive( 100 ))
-                received = true;
+            {
+#ifdef ZEQ_USE_ZEROBUF
+                received = echo.gotData && received;
+#endif
+            }
         }
     }
```
Comment (eile) [inline tests/connection/broker.cpp]: Shouldn't this be two separate tests? Right now the FB code path is not tests as far as I understand.

File: tests/connection/broker.cpp [hunk 2]
```diff
@@ -70,6 +78,12 @@ class Subscriber
     mutable std::mutex _mutex;
     bool _started;
 
+    void onEchoEvent( const zeq::Event& event )
+    {
+        test::onEchoEvent( event );
+        received = true;
+    }
+
```
Comment (eile) [inline tests/connection/broker.cpp]: Ah, here. hmm.

File: zeq/subscriber.h [hunk 3]
```diff
@@ -73,6 +73,27 @@ class Subscriber : public Receiver
      */
     ZEQ_API bool deregisterHandler( const uint128_t& event );
 
+    /**
+     * Subscribe a ZeroBuf object to updates published by any connected
+     * receiver.
```
Comment (eile) [inline zeq/subscriber.h]: Subscribe a ZeroBuf object to receive updates from any connected publisher

File: zeq/subscriber.h [hunk 4]
```diff
@@ -73,6 +73,27 @@ class Subscriber : public Receiver
...
+     *
+     * Every received update will directly applied on the ZeroBuf object. To
```
Comment (eile) [inline zeq/subscriber.h]: Every update will be directly applied on the ZeroBuf object during receive().

Comment (eile) [inline zeq/subscriber.h]: The subscribed object instance has to be valid until unsubscribe().

**Comment by eile** [comment]
+1 otherwise.

**Pull request merged**